rfd/0181-pg-database-access-webui.md

```diff
@@ -0,0 +1,322 @@
+her queries.
+
+![PostgreSQL interactive shell](assets/0181-pg-shell.png)
+
+After interacting with the database, she closes the tab, and her database
+session ends.
+
+##### Auto-user provisioning enabled
+
+This is the same scenario, but the PostgreSQL instance was configured with user
+provisioning enabled. This change implies which information Alice sees on the
+connect modal. She doesn’t need to select the database user, as it will default
+to her username. The select is then disabled, and a new now select will be
+presented where she can select database roles attached to their user.
+
+![PostgreSQL connect modal with database roles](assets/0181-connect-dialog-roles.png)
+
+#### User story: Bob access database with per-session MFA enabled.
+
+Bob is familiar with Teleport web UI and is experienced with PostgreSQL,
+spending considerable time using `psql`, the PostgreSQL CLI.
+
+The system administrators configured this Teleport cluster to use MFA per
+session (`require_session_mfa: true`). They also have granted Bob permission to
+connect to multiple PostgreSQL databases. The role he'll be using has a short
+session TTL.
+
+He logs into Teleport web UI, searches for the desired database, and clicks
+"Connect". He’s presented with a pre-filled form containing his connection
+information: database name, database roles, and database user. He confirms the
+values and clicks "Connect" to start a new session. He's then presented with a
+tab browser tab with the PostgreSQL interactive shell.
+
+In the new tab, he's prompted with the MFA modal (the same as SSH
+sessions). After completing the authentication, he starts performing multiple
+queries using the interactive shell, but after some time, he exceeds his session
+TTL, the connection is dropped, and he is redirected to the login page. His
+database session is terminated.
+
+#### PostgreSQL interactive terminal
+
+##### Banner
+
+At the beginning of each session, we'll include an informative banner that will consist of the following:
+- Information about the interactive shell name and its version.
+- Connected PostgreSQL version (this mimics the `psql`).
+- Parameters used to connect to the instance, like the database and username.
+- Short text describing how to get the supported commands and information through the help command (more details in the next section).
+
+```shell
+Teleport PostgreSQL interactive shell (v16.2.0-dev)
+Connected to "pg" instance (16.2) as "alice" user (with roles "read-write").
+Type "help" or \? for help.
+```
+
+##### Supported commands
+
+In addition to executing queries, the interactive shell will implement some
+backlash (\\) commands from `psql`. Users can fetch the list of supported
+commands by calling the help command (`\?` or `help`):
+
+```shell
+postgres=> \?
+General:
+  \q          Terminate the session.
+  \teleport   Show Teleport interactive shell information, such as execution limitations.
```

Comment:
could you share some samples on `\teleport` and `\session`? Depending on what `\teleport` shows, they might just fit in `help` instead of a separate command. Is `\session` similar to `\conninfo` from `psql`?